src/Common/src/CoreLib/System/Numerics/Vector.cs

```diff
@@ -1301,89 +1301,57 @@ public bool Equals(Vector<T> other)
         /// <returns>The hash code.</returns>
         public override int GetHashCode()
         {
-            int hash = 0;
-
             if (Vector.IsHardwareAccelerated)
             {
                 if (typeof(T) == typeof(Byte))
                 {
+					var hash = new HashCode();
                     for (int g = 0; g < Count; g++)
                     {
-                        hash = HashHelpers.Combine(hash, ((Byte)(object)this[g]).GetHashCode());
+                        hash.Add((Byte)(object)this[g]);
```

Comment:
Are these casts still necessary?  `HashCode.Add` takes a `T`, so `hash.Add(this[g])` should be fine, right?